util/src/test/java/com/google/cloud/hadoop/util/GoogleCredentialWithIamAccessTokenTest.java

```diff
@@ -0,0 +1,66 @@
+package com.google.cloud.hadoop.util;
+
+import static com.google.cloud.hadoop.util.testing.MockHttpTransportHelper.jsonDataResponse;
+import static com.google.cloud.hadoop.util.testing.MockHttpTransportHelper.mockTransport;
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.testing.http.MockHttpTransport;
+import com.google.api.client.util.Clock;
+import com.google.api.services.iamcredentials.v1.model.GenerateAccessTokenResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public class GoogleCredentialWithIamAccessTokenTest {
```

Comment:
Add JavaDoc

Reply:
We use default value of 3600s for now, which is the longest the IAM access token is valid for. If for some reason the server didn't set the expirationValue the token will just be valid for whatever value set in request.